pkg/controller/render/render_controller.go

```diff
@@ -481,6 +481,23 @@ func (ctrl *Controller) syncGeneratedMachineConfig(pool *mcfgv1.MachineConfigPoo
 		return err
 	}
 
+	// For now before we generate the rendered configs, let's also update the MCs here
+	// TODO should move this
+	for _, config := range configs {
+
+		needUpdate, err := ctrlcommon.NeedUpdateConfigToSpec3(config)
+		if err != nil {
+			return err
+		}
+		if needUpdate {
+			_, err = ctrl.client.MachineconfigurationV1().MachineConfigs().Update(context.TODO(), config, metav1.UpdateOptions{})
```

Comment:
For non-rendered configs I probably need a better location to update them, there is also the Boostrap path to consider

Reply:
that said with this update in place we will no longer be able to rollback, so maybe for user-created configs I should keep it in the version that it was on?